src/visitors/rules/oas2-structural/validateOAS2Parameter.js

```diff
@@ -87,90 +119,132 @@ class ValidateOAS2Parameter {
 
       pattern(node, ctx) {
         if (node.in && node.in === 'body' && node.pattern) {
-          return ctx.createError('This field is allowed only if the parameter not in "body". In such case, use Schema object instead');
+          return ctx.createError(
+            'This field is allowed only if the parameter not in "body". In such case, use Schema object instead',
+          );
+        }
+        if (node && node.pattern && typeof node.pattern !== 'string') {
+          return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('string'), 'value');
         }
-        if (node && node.pattern && typeof node.pattern !== 'string') return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('string'), 'value');
         return null;
       },
       maximum(node, ctx) {
         if (node.in && node.in === 'body' && node.maximum) {
-          return ctx.createError('This field is allowed only if the parameter not in "body". In such case, use Schema object instead');
+          return ctx.createError(
+            'This field is allowed only if the parameter not in "body". In such case, use Schema object instead',
+          );
+        }
+        if (node && node.maximum && typeof node.maximum !== 'number') {
+          return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('number'), 'value');
         }
-        if (node && node.maximum && typeof node.maximum !== 'number') return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('number'), 'value');
         return null;
       },
       minimum(node, ctx) {
         if (node.in && node.in === 'body' && node.minimum) {
-          return ctx.createError('This field is allowed only if the parameter not in "body". In such case, use Schema object instead');
+          return ctx.createError(
+            'This field is allowed only if the parameter not in "body". In such case, use Schema object instead',
+          );
+        }
+        if (node && node.minimum && typeof node.minimum !== 'number') {
+          return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('number'), 'value');
         }
-        if (node && node.minimum && typeof node.minimum !== 'number') return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('number'), 'value');
         return null;
       },
       maxLength(node, ctx) {
         if (node.in && node.in === 'body' && node.maxLength) {
-          return ctx.createError('This field is allowed only if the parameter not in "body". In such case, use Schema object instead');
+          return ctx.createError(
+            'This field is allowed only if the parameter not in "body". In such case, use Schema object instead',
```

Comment:
I'll stop leaving the suggestions, and I'm wondering if this should be a string constant/variable so that we don't have to repeat it everywhere and can adjust it in one place instead?